plugins/modem-manager/fu-mm-device.c

```diff
@@ -1504,6 +1546,35 @@ fu_mm_device_attach(FuDevice *device, FuProgress *progress, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_mm_device_setup(FuDevice *device, GError **error)
+{
+	FuMmDevice *self = FU_MM_DEVICE(device);
+	g_autoptr(FuDeviceLocker) locker = NULL;
+	g_autoptr(GError) error_branch = NULL;
+
+	/* Create IO channel to send AT commands to the modem */
+	locker = fu_device_locker_new_full(device,
+					   (FuDeviceLockerFunc)fu_mm_device_io_open,
+					   (FuDeviceLockerFunc)fu_mm_device_io_close,
+					   &error);
```

Comment:
you need to either check locker for NULL here (and return FALSE) or use something like error_locker and then only do the at_cmd when locker != NULL. Also, no & on the error.